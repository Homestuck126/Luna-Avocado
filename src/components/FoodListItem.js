import React from "react";
import { View, Text, StyleSheet, Pressable } from "react-native";

const FoodListItem = ({ data, onPress }) => {
  return (
    <Pressable
      onPress={() => onPress(data)}
      style={({ pressed }) => [
        {
          backgroundColor: pressed
            ? "rgba(220,220,230,.3)"
            : "rgb(248,248,255)",
        },
        styles.container,
      ]}
    >
      <Text style={styles.name}>{data.name}</Text>
      <Text style={styles.macros}>Calories: {data.macros[0]}</Text>
      <Text style={styles.macros}>Protein: {data.macros[1]}</Text>
      <Text style={styles.macros}>Carbs: {data.macros[2]}</Text>
      <Text style={styles.macros}>Fat: {data.macros[3]}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 120,
    width: "95%",
    alignItems: "flex-start",
    borderColor: "#C0C0C0",
    borderWidth: 3,
    borderRadius: 10,
    padding: 10,
    margin: 10,
  },
  name: {
    fontSize: 18,
    fontWeight: "bold",
    marginBottom: 5,
  },
  macros: {
    fontSize: 14,
    color: "#666",
  },
});

export default FoodListItem;
