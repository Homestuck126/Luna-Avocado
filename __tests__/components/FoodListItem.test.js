import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import FoodListItem from "../../src/components/FoodListItem";

describe("FoodListItem", () => {
  it("calls onPress when isPressable == true", () => {
    const mockOnPress = jest.fn();
    const mockData = {
      id: null,
      name: "test food",
      macros: [2000, 100, 90, 80],
    };

    const { getByTestId } = render(
      <FoodListItem
        data={mockData}
        onPress={mockOnPress}
        isPressable={true}
      ></FoodListItem>
    );

    const foodListItemPressable = getByTestId("FoodListItem:ItemPressable");

    fireEvent.press(foodListItemPressable);
    expect(mockOnPress).toHaveBeenCalled();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("displays same info from data in Text components", () => {
    const mockData = {
      id: null,
      name: "test food",
      macros: [2000, 100, 90, 80],
    };

    render(<FoodListItem data={mockData} onPress={() => {}}></FoodListItem>);

    expect(screen.getByText("test food")).toHaveTextContent("test food");
    expect(
      screen.getByText(
        "Calories",
        (options = {
          exact: false,
        })
      )
    ).toHaveTextContent("Calories: 2000");
    expect(
      screen.getByText(
        "Protein",
        (options = {
          exact: false,
        })
      )
    ).toHaveTextContent("Protein: 100");
    expect(
      screen.getByText(
        "Carbs",
        (options = {
          exact: false,
        })
      )
    ).toHaveTextContent("Carbs: 90");
    expect(
      screen.getByText(
        "Fat",
        (options = {
          exact: false,
        })
      )
    ).toHaveTextContent("Fat: 80");
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
