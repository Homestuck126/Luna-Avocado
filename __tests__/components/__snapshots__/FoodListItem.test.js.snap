// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodListItem calls onPress when isPressable == true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "rgb(248,248,255)",
      },
      {
        "alignItems": "flex-start",
        "borderColor": "#C0C0C0",
        "borderRadius": 10,
        "borderWidth": 3,
        "height": 120,
        "margin": 10,
        "padding": 10,
        "width": "95%",
      },
    ]
  }
  testID="FoodListItem:ItemPressable"
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    test food
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Calories: 
    2000
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Protein: 
    100
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Carbs: 
    90
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Fat: 
    80
  </Text>
</View>
`;

exports[`FoodListItem displays same info from data in Text components 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "rgb(248,248,255)",
      },
      {
        "alignItems": "flex-start",
        "borderColor": "#C0C0C0",
        "borderRadius": 10,
        "borderWidth": 3,
        "height": 120,
        "margin": 10,
        "padding": 10,
        "width": "95%",
      },
    ]
  }
  testID="FoodListItem:ItemPressable"
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    test food
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Calories: 
    2000
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Protein: 
    100
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Carbs: 
    90
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
      }
    }
  >
    Fat: 
    80
  </Text>
</View>
`;
